test/e2e/gardener/shoot/shoot.go

```diff
@@ -248,6 +259,28 @@ func ItShouldInitializeSeedClient(s *ShootContext) {
 	}, SpecTimeout(time.Minute))
 }
 
+// ItShouldComputeControlPlaneNamespace computes the control plane namespace for the shoot and stores it in ShootContext.ControlPlaneNamespace.
+// This is achiaved via `ComputeTechnicalID`. The namespace is constructed from the project name and the shoot name.
+// It's hard to determine what the namespace of the control-plane will be before shoot creation, thus to compute it we need to first create the Shoot.
+func ItShouldComputeControlPlaneNamespace(s *ShootContext) {
+	GinkgoHelper()
+
+	It("Compute Control Plane Namespace", func(ctx SpecContext) {
+		ns := &corev1.Namespace{}
+
+		err := s.GardenClient.Get(ctx, client.ObjectKey{Name: s.Shoot.Namespace}, ns)
+		Expect(err).NotTo(HaveOccurred(), "could not get namespace %q in Garden cluster: %w", s.Shoot.Namespace, err)
+		Expect(ns.Labels).NotTo(BeEmpty(), "namespace %q does not have any labels", ns.Name)
+
+		projectName, ok := ns.Labels[v1beta1constants.ProjectName]
+		Expect(ok).To(BeTrue(), "namespace %q did not contain a project label", ns.Name)
+
+		namespace := gardenerutils.ComputeTechnicalID(projectName, s.Shoot)
```

Comment:
Why do you compute this yourself? It is in the `.status.technicalID` of the `Shoot` - just use it from there?